ipmipower/ipmipower_connection.c

```diff
@@ -335,29 +334,78 @@ _connection_setup (struct ipmipower_connection *ic, const char *hostname)
   strncpy (ic->hostname, hostname_second_parse_ptr, MAXHOSTNAMELEN);
   ic->hostname[MAXHOSTNAMELEN] = '\0';
 
-  /* Determine the destination address */
-  bzero (&(ic->destaddr), sizeof (struct sockaddr_in));
-  ic->destaddr.sin_family = AF_INET;
-  ic->destaddr.sin_port = htons (port);
-
-  if (!(result = gethostbyname (ic->hostname)))
+  snprintf(port_str, sizeof (port_str), "%d", port);
+  memset(&ai_hints, 0, sizeof (struct addrinfo));
+  ai_hints.ai_family = AF_UNSPEC;
+  ai_hints.ai_socktype = SOCK_DGRAM;
+  ai_hints.ai_flags = (AI_V4MAPPED | AI_ADDRCONFIG);
+  if ((result = getaddrinfo (ic->hostname, port_str, &ai_hints, &ai_res )) < 0)
     {
       if (h_errno == HOST_NOT_FOUND)
         ipmipower_output (IPMIPOWER_MSG_TYPE_HOSTNAME_INVALID, ic->hostname, NULL);
       else
         {
 #if HAVE_HSTRERROR
-          IPMIPOWER_ERROR (("gethostbyname() %s: %s", ic->hostname, hstrerror (h_errno)));
+          IPMIPOWER_ERROR (("getaddrinfo() %s: %s", ic->hostname, hstrerror (h_errno)));
```

Comment:
on my redhat box, it says it should be ```gai_strerror```, not ```hstrerror```?  Unsure if they map to the same thing or not.  I suppose a ```#if``` equivalent would have to be done.  Portability of ```gai_strerror``` personally I'm not knowledgeable about.

I suppose other "h_errno" dealings around here too.